Assets/Scripts/Player/PlayerMovement.cs

```diff
@@ -0,0 +1,27 @@
+using UnityEngine;
+
+public sealed class PlayerMovement : MonoBehaviour
+{
+    [SerializeField] private float movingSpeed;
+    
+    private CharacterController _characterController;
+    private Vector2 _movementInput;
+    private Vector2 _moveDirection;
+
+    private void Awake() => _characterController = GetComponent<CharacterController>();
+
+    private void FixedUpdate() => SetMoveDirection();
+
+    private void GetInput(out Vector2 input2D)
+    {
+        input2D = Vector2.zero;
+        input2D = new Vector2(Input.GetAxisRaw("Horizontal"), Input.GetAxisRaw("Vertical"));

```

Comment:
```suggestion
        input2D.x = Input.GetAxisRaw("Horizontal");
        input2D.y = Input.GetAxisRaw("Vertical");
```
Omdat je nu de variable aanmaakt als parameter, hoef je hem niet eerst als Vector2.zero neer te zetten.

Je wilt alleen ook niet elk frame een nieuwe Vector2 toevoegen aan de Heap, dit gaat uiteindelijk slecht zijn voor performance. Dit kan voorkomen worden door de x and y apart te setten.

Het mooie van deze setup is dat je geen extra member variables nodig hebt, geen extra local variables hoeft aan te maken en niet eens een return type hoeft te geven aan de functie! 

(Hoop dat de comment goed staat, ik zit op telefoon)